src/domains/ArbitrumDomain.sol

```diff
@@ -46,45 +46,56 @@ contract ArbSysOverride {
 
 contract ArbitrumDomain is BridgedDomain {
 
-    InboxLike public immutable inbox;
-    address public immutable arbSys;
-    BridgeLike public immutable bridge;
+    InboxLike  public inbox;
+    address    public arbSys;
+    BridgeLike public bridge;
 
     address public l2ToL1Sender;
 
     bytes32 constant MESSAGE_DELIVERED_TOPIC = keccak256("MessageDelivered(uint256,bytes32,address,uint8,address,bytes32,uint256,uint64)");
     bytes32 constant SEND_TO_L1_TOPIC = keccak256("SendTxToL1(address,address,bytes)");
 
     constructor(string memory _config, string memory _name, Domain _hostDomain) Domain(_config, _name) BridgedDomain(_hostDomain) {
-        inbox = InboxLike(readConfigAddress("inbox"));
-        arbSys = readConfigAddress("arbSys");
-        bridge = BridgeLike(inbox.bridge());
-        vm.recordLogs();
-
-        // Make this contract a valid outbox
-        address _rollup = bridge.rollup();
-        vm.store(
-            address(bridge),
-            bytes32(uint256(8)),
-            bytes32(uint256(uint160(address(this))))
-        );
-        bridge.setOutbox(address(this), true);
-        vm.store(
-            address(bridge),
-            bytes32(uint256(8)),
-            bytes32(uint256(uint160(_rollup)))
-        );
-
-        // Need to replace ArbSys contract with custom code to make it compatible with revm
-        uint256 fork = vm.activeFork();
-        selectFork();
-        bytes memory bytecode = vm.getCode("ArbitrumDomain.sol:ArbSysOverride");
-        address deployed;
-        assembly {
-            deployed := create(0, add(bytecode, 0x20), mload(bytecode))
+        string memory rpcEnv = readConfigString("rpc");
+        string memory rpc = vm.envString(rpcEnv);
+        if (bytes(rpc).length > 0) {
```

Comment:
I'd propose changing to:
```
string memory rpc = vm.envString(readConfigString("rpc"));
if (bytes(rpc).length == 0) return;
```

Less verbosity and this way we can also reduce the changes in the PR as the rest of the code gets back to the original indentation.

Reply:
Same for Optimism.

Reply:
TBH I would just revert this whole section and replace with the rpc being defined in the constructor. You shouldn't be instantiating a domain instance if it's not attached to an RPC. I don't like the silent failure mode here.

Reply:
Yeah makes sense to revert if rpc is not present.

Reply:
> TBH I would just revert this whole section and replace with the rpc being defined in the constructor. You shouldn't be instantiating a domain instance if it's not attached to an RPC. I don't like the silent failure mode here.

I added this so that we don't always have to instantiate RPC for optimism and arbitrum in spells all the time.  I was trying to make it a seamless as possible for spell crafters to add/remove the L2 testing depending on whether the spell had an L2 component.  If this reverts then they will have to edit setup code to enable or disable l2 testing so i felt it was easier to handle that just with ENVs.  

Further I was thinking of a situation where we eventually have a systems value test for l2s.  The spell might not change anything on the L2s but we want CI or the full test suite to test L2 MCD values in the same way we always test L1 system values.  In this instance, if this reverts then whoever is running the test suite will always have to have all L2 RPC values set or they are unable to run things.  With this `return` option you could test just L1 by only exporting `ETH_RPC_URL` and not have to edit test files or try to come up with the correct regex for `match` to only get L1 tests.

Reply:
I hear the use case, but I think this should be dealt with by users of this library (IE in spells). If you want to de-activate part of the tests (IE L2 part) then the test harness should not be instantiating `ArbitrumDomain`, etc.